### ISSUE TITLE
Terraform Output coming as a Json

### ISSUE THREAD

**Issue by ma-kadri** [gruntwork-io/terratest#737]
I used to run  something like this to fetch terraform outputs and run some comparison on top of it, 

```
	// Checking EFS file system is created.
	efsArn := terraform.Output(t, terraformOptions, "cluster_efs_arn")
```

It used to provide a output like this ,

```
TestTerraformAwsEcs 2020-11-25T13:16:05Z retry.go:91: terraform [output -no-color cluster_efs_arn]
2280TestTerraformAwsEcs 2020-11-25T13:16:05Z logger.go:66: Running command terraform with args [output -no-color cluster_efs_arn]
2281TestTerraformAwsEcs 2020-11-25T13:16:06Z logger.go:66: arn:aws:elasticfilesystem:us-east-1:21354357:file-system/fs-7ec38b
```

now all of a sudden i see a change, i.e it started to provide output as json for no reason. 

```
TestTerraformAwsEcs 2020-12-13T16:00:06Z logger.go:66: Running command terraform with args [output -no-color -json cluster_efs_arn]
TestTerraformAwsEcs 2020-12-13T16:00:11Z logger.go:66: {
TestTerraformAwsEcs 2020-12-13T16:00:11Z logger.go:66:     "sensitive": false,
TestTerraformAwsEcs 2020-12-13T16:00:11Z logger.go:66:     "type": "string",
TestTerraformAwsEcs 2020-12-13T16:00:11Z logger.go:66:     "value": "arn:aws:elasticfilesystem:us-east-1:213543574:file-system/fs-f10d7904"
TestTerraformAwsEcs 2020-12-13T16:00:11Z logger.go:66: }
```
i have not made any change either in terraform or in terratest,, how can i get rid of this json for permanent basis

**Comment by brikis98** [comment]
What version of Terratest are you using? We intentionally changed how `terraform.Output` works under the hood to use `-json` due to a backwards incompatible change in Terraform itself. See https://github.com/gruntwork-io/terratest/pull/719 for details. This was released in https://github.com/gruntwork-io/terratest/releases/tag/v0.31.0 and then patched in https://github.com/gruntwork-io/terratest/releases/tag/v0.31.1. That said, while what's happening under the hood is different, the external behavior of the function should be unchanged: i.e., `terraform.Output` should return the same value as before. Are you not seeing that?

**Comment by ma-kadri** [comment]
so i was using the lastest one (0.31.1), i.e i had not locked its version before,,so this was the one that broker it some 5 to 6 days back. Now i had to downgrade it  manually to below version to unblock my production realease.

And Nope, now the default behaviour is json output as shared above, which fails my test,,,  is there any way that we can use the latest version and still get json less output

```
[[override]]
  name = "github.com/gruntwork-io/terratest"
  version = "v0.30.8"
```

**Comment by brikis98** [comment]
> so i was using the lastest one (0.31.1), i.e i had not locked its version before,,so this was the one that broker it some 5 to 6 days back. Now i had to downgrade it manually to below version to unblock my production realease.

Yea, we always recommend pinning a version to avoid accidentally pulling in backwards incompatible changes.s

> And Nope, now the default behaviour is json output as shared above, which fails my test,,, is there any way that we can use the latest version and still get json less output

Can you share more details on what you're seeing? That is, your Terraform code with the `output` and your Terratest code that calls `terraform.Output`?

**Comment by ma-kadri** [comment]
so i do some thing like this to set some variables after terraform succeeds.

```
	// Checking ALB & Other services are UP and RUNNING.
	albDNS := terraform.Output(t, terraformOptions, "alb_url")
	flowerDNS := terraform.Output(t, terraformOptions, "flower_dns_name")
```

And then do a simple endpoint test of these services.

```
	http_helper.HttpGetWithRetryWithCustomValidation(t, fmt.Sprintf("http://%s", albDNS), &tlsConfig, retries, SleepBetweenRetries, func(statusCode int, body string) bool {
		if statusCode == albCode {
			logger.Logf(t, "Got expected status code %d from URL http://%s", albCode, albDNS)
			return true
		}
		return false
	})
```

**Comment by brikis98** [comment]
And what happens when you run that code? The JSON output in the logs is expected... That's the output `terraform output` returns when you pass the `-json` flag. The `terraform.Output` method should be taking that JSON output and parsing it into a string for you, so the value returned for `albDNS` and `flowerDNS` should be the exact output strings you expect.

**Comment by ma-kadri** [comment]
That's where the problem lies,, I don't need the json output,, and I have not made the change to get the json output i.e -json not set by me,,, it's automatically picking up -json with the new update,,, 

Now for me to use it as is, I would need to add more logic later to extract the value,, so that I can run the http testing,,

So the main question can we or do we have something which can eliminate the json output,, like passing flag or something

**Comment by brikis98** [comment]
Can you paste the EXACT output you're getting from `terraform.Output`? Not what you see in the logs, but the value of the value returned. E.g., Print out `albDNS` and `flowerDNS` in your example above.

**Comment by GreenyMcDuff** [comment]
Hey, I'm experiencing a similar issue to this.  I think the terraform.output function is returning the command rather than the value of the output variable.

I've raised a [question in StackOverflow.](https://stackoverflow.com/questions/65391466/terratest-assert-string-fails-with-github-actions)  Would it be helpful for me to raise a separate issue in GitHub or do you think it's similar enough to track here?

**Comment by GreenyMcDuff** [comment]
Ignore my comment, the solution proposed [in this issue](https://github.com/gruntwork-io/terratest/issues/706) solved it for me

**Comment by infraredgirl** [comment]
Since we haven't seen any recent activity here, I'll assume the issue has been solved and close this ticket. However @ma-kadri please feel free to reopen if you still need help with this!

**Comment by howdoicomputer** [comment]
I'm coming across this same issue. The `output` function isn't parsing the JSON and just returning the string as is.

**Comment by yorinasub17** [comment]
Hi @howdoicomputer , can you open a new issue, with the following details?

- version of terratest
- version of terraform
- terratest go code snippet for the relevant call to `Output`
- relevant log snippets that you can share
